### PR TITLE
Make fuzz task work on a uworker.

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This has been verified locally. This involved:
* Get data bundle directory in an untrusted manner.
* Upload fuzzing engine output in an untrusted manner.
* Not checking for last corpus sync on a uworker. This is a trusted operation and is uneeded because uworkers do not outlive the task and thus will never have the corpus.
* Refactor upload_log so callers aren't exposed to its internals.

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/tasks/setup.py [hunk 1]
```diff
@@ -744,17 +744,15 @@ def _is_data_bundle_up_to_date(data_bundle, data_bundle_directory):
...
-  """For fuzz_task which doesn't get data bundles in an untrusted manner."""
-  # TODO(metzman): Delete this when fuzz_task is migrated.
-  # Check if we have a fuzzer-specific data bundle. Use it to calculate the
-  # data directory we will fetch our testcases from.
-  data_bundle = data_types.DataBundle.query(
-      data_types.DataBundle.name == fuzzer.data_bundle_name).get()
-  return get_data_bundle_directory(fuzzer, data_bundle)
+def get_data_bundle_directory(fuzzer, setup_input):
+  assert setup_input.data_bundle_corpuses[0].data_bundle
+  data_bundle = setup_input.data_bundle_corpuses[0].data_bundle
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/tasks/setup.py:749]: Add a comment to explain why we are only doing [0] ?

File: src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py [hunk 2]
```diff
@@ -548,7 +550,8 @@ def sync_from_gcs(self):
 
     # Check if the corpus was recently synced. If yes, set a flag so that we
     # don't sync it again and save some time.
-    if last_sync_time and os.path.exists(self._corpus_directory):
+    if not environment.is_uworker() and last_sync_time and os.path.exists(
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py:553]: add a TODO to completely remove this code once we are completely migrated?

**Review by oliverchang — APPROVED** [review]

File: src/clusterfuzz/_internal/bot/tasks/setup.py (hunk 1 above)
Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/bot/tasks/setup.py:749]: Done.

File: src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py (hunk 2 above)
Comment (jonathanmetzman) [reply src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py:556]: Done.

**Pull request merged**